pkg/querier/tenantfederation/merge_queryable.go

```diff
@@ -80,47 +103,41 @@ type MergeQuerierCallback func(ctx context.Context, mint int64, maxt int64) (ids
 // If the label `idLabelName` is already existing, its value is overwritten and
 // the previous value is exposed through a new label prefixed with "original_".
 // This behaviour is not implemented recursively.
-func NewMergeQueryable(idLabelName string, callback MergeQuerierCallback, byPassWithSingleQuerier bool, maxConcurrency int, logger log.Logger) storage.Queryable {
+func NewMergeQueryable(idLabelName string, callbacks MergeQuerierCallbacks, bypassWithSingleID bool, maxConcurrency int, logger log.Logger) storage.Queryable {
 	return &mergeQueryable{
-		logger:                  logger,
-		idLabelName:             idLabelName,
-		callback:                callback,
-		bypassWithSingleQuerier: byPassWithSingleQuerier,
-		maxConcurrency:          maxConcurrency,
+		logger:             logger,
+		idLabelName:        idLabelName,
+		callbacks:          callbacks,
+		bypassWithSingleID: bypassWithSingleID,
+		maxConcurrency:     maxConcurrency,
 	}
 }
 
 type mergeQueryable struct {
-	logger                  log.Logger
-	idLabelName             string
-	bypassWithSingleQuerier bool
-	callback                MergeQuerierCallback
-	maxConcurrency          int
+	logger             log.Logger
+	idLabelName        string
+	bypassWithSingleID bool
+	callbacks          MergeQuerierCallbacks
+	maxConcurrency     int
 }
 
 // Querier returns a new mergeQuerier, which aggregates results from multiple
 // underlying queriers into a single result.
-func (m *mergeQueryable) Querier(ctx context.Context, mint int64, maxt int64) (storage.Querier, error) {
-	// TODO: it's necessary to think how to override context inside querier
+func (m *mergeQueryable) Querier(mint int64, maxt int64) (storage.Querier, error) {
+	// TODO: it's necessary to think of how to override context inside querier
 	//  to mark spans created inside querier as child of a span created inside
 	//  methods of merged querier.
-	ids, queriers, err := m.callback(ctx, mint, maxt)
+	upstream, err := m.callbacks.Querier(mint, maxt)
```

Comment:
Actually I am unsure. I was going to confer with @charleskorn, since I think it could be related to something he tested this PR for.

Reply:
I didn't check queries with tenant federation specifically. However, I'd expect that this comment is no longer valid: we no longer need to do anything special to make sure spans are correctly linked to their parents because we can pass the context from the parent to the child.

Reply:
I've just tried a query with tenant federation and the traces look correct to me.

Reply:
Thank you, I've removed the TODO comment!